qemu/tests/dump_guest_memory.py

```diff
@@ -0,0 +1,229 @@
+                if strict_match:
+                    if operator.eq(key, element):
+                        return True
+                else:
+                    if key in str(element):
+                        return True
+        return False
+
+    def _check_dict(dic, key, val, check_item_in_pair=True):
+        """
+        Check if the expect key, val are contained in QMP output dic.
+
+        :param dic: content of QMP command return value
+        :type dic: dict
+        :param key: expect result
+        :type key: str
+        :param val: expect result
+        :type val: str or None(if check_item_in_pair=False)
+        :param check_item_in_pair: If expect result is dict or str
+        :type check_item_in_pair: bool. Means expect result is dict or str.
+
+        :return check result
+        :rtype: bool
+        """
+        if key in dic and not check_item_in_pair:
+            return True
+        elif key in dic and val == dic[key]:
+            return True
+        else:
+            for value in dic.values():
+                if isinstance(value, dict):
+                    if _check_dict(value, key, val, check_item_in_pair):
+                        return True
+                elif isinstance(value, list):
+                    if check_list(value, key, val, check_item_in_pair):
+                        return True
+                elif value != '' and not check_item_in_pair:
+                    if strict_match:
+                        if operator.eq(key, value):
+                            return True
+                    else:
+                        if key in str(value):
+                            return True
+            return False
+
+    def check_result(qmp_o, expect_o=None):
+        """
+        Check test result with difference way according to result_check.
+        result_check = equal, expect_o should equal to qmp_o.
+        result_check = contain, expect_o should be contained in qmp_o
+        result_check = not_contain, expect_o should not be contained in qmp_o.
+
+        :param qmp_o: output from qmp_cmd.
+        :type qmp_o: list
+        :param expect_o: the expect result.
+        :type expect_o: list
+        """
+        logging.info("Expect result is %s" % expect_o)
+        logging.info("Actual result that get from qmp_cmd is %s" % qmp_o)
+        result = None
+        if result_check == "equal":
+            if not operator.eq(qmp_o, expect_o):
+                test.fail("QMP output does not equal to the expect result.\n "
+                          "Expect result: '%s'\n"
+                          "Actual result: '%s'" % (expect_o, qmp_o))
+        elif result_check == "contain":
+            for o in expect_o:
+                if isinstance(o, dict):
+                    for key, val in o.items():
+                        result = check_list(qmp_o, key, val)
+                        if not result:
+                            break
+                elif isinstance(o, str):
```

Comment:
I see all `cmd_return_value` is dictionary, if no other types, I recommend not to make your script so complicated.